src/shared/resourcefetcher/resourcefetcher.ts

```diff
@@ -3,6 +3,7 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+// TODO: this is just a "thunk". Replace it with something more generic.
```

Comment:
I like to call these interfaces 'asynchronous pointers'. I see them all the time. Of course in this case it is also used as a thunk.